fortran77-rules/src/main/resources/lex/COMPRESFileLength.lex

```diff
@@ -0,0 +1,159 @@
+/************************************************************************************************/
+/* i-Code CNES is a static code analyzer.                                                       */
+/* This software is a free software, under the terms of the Eclipse Public License version 1.0. */ 
+/* http://www.eclipse.org/legal/epl-v10.html                                                    */
+/************************************************************************************************/ 
+
+/********************************************************************************/
+/* This file is used to generate a rule checker for COM.PRES.FileLength rule.   */
+/* For further information on this, we advise you to refer to RNC manuals.      */
+/* As many comments have been done on the ExampleRule.lex file, this file       */
+/* will restrain its comments on modifications.                                 */
+/*                                                                              */
+/********************************************************************************/
+package fr.cnes.icode.fortran77.rules;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.File;
+import java.util.List;
+import java.util.LinkedList;
+import fr.cnes.icode.data.AbstractChecker;
+import fr.cnes.icode.data.CheckResult;
+import fr.cnes.icode.exception.JFlexException;
+%%
+%class COMPRESFileLength
+%extends AbstractChecker
+%public
+%column
+%line
+%function run
+%yylexthrow JFlexException
+%type List<CheckResult>
+%state COMMENT, NAMING, NEW_LINE, LINE, AVOID, INLINE_COMMENT
+COMMENT_WORD = \!         | c          | C     | \*
+FREE_COMMENT = \!
+FUNC         = FUNCTION   | function
+PROC         = PROCEDURE  | procedure
+SUB          = SUBROUTINE | subroutine
+PROG         = PROGRAM    | program
+MOD          = MODULE     | module
+INTER        = INTERFACE  | interface
+TYPE         = {FUNC}     | {PROC}     | {SUB} | {PROG} | {MOD} | {INTER}
+VAR          = [a-zA-Z][a-zA-Z0-9\_]*
+CLOSING      = END[\ ]*IF | end[\ ]*if | END[\ ]*DO | end[\ ]*do
+END          = END        | end
+STRING       = \'[^\']*\' | \"[^\"]*\"
+SPACE        = [\ \r\t\f]
+BLANK_LINE 	 = {SPACE}*\R 
+                                                                
+%{
+    String location = "MAIN PROGRAM";
+    private String parsedFileName;
+    int codeLines = 0;
+    int numTotal = 1;
```

Comment:
The implementation calculated lines of code and lines in general (including blanks and comments) so that a threshold could be checked against both of them. At request of EUMETSAT, final implementation of the rule only took into account lines of code, so numTotal is not actually used.
We have eliminated the variable numTotal in a new commit.